tests/integration/targets/postgresql_ping/tasks/postgresql_ping_initial.yml

```
@@ -46,6 +46,39 @@
       - result.server_version == {}
       - result is not changed
 
+- name: postgresql_ping - check ping of the database on non-existent port does not return anything
+  become_user: "{{ pg_user }}"
+  become: true
+  environment:
+    PGPORT: 5435
+  ignore_errors: true
+  postgresql_ping:
+    db: "{{ db_default }}"
+    login_user: "{{ pg_user }}"
+  register: result
+
+- assert:
+    that:
+      - result.is_available == false
+      - result.server_version == {}
+      - result is not changed
+
+- name: postgresql_ping - check ping of the database by a non-existent user does not return anything
+  become_user: "{{ pg_user }}"
+  become: true
+  environment:
+    PGUSER: 'test_user'
```

```suggestion
    PGUSER: 'test_user'
```
how about adding another task with setting `login_user: "{{ pg_user }}"` via `environment` and checking it returns that the server is available?